### PR TITLE
DO NOT MERGE -- for review only

### PR DESCRIPTION
**Pull request by inantop**

### CODE REVIEW

File: easy_profiler_core/CMakeLists.txt [hunk 1]
```diff
@@ -20,9 +18,8 @@ if (NOT WIN32)
             set(NO_CXX11_THREAD_LOCAL_SUPPORT TRUE)
         endif ()
     endif ()
-    
     # TODO: Check thread_local keyword support for other compilers for Unix
-    
+
```
Comment (mathieumallet) [inline easy_profiler_core/CMakeLists.txt]: unnecessary whitespace changes :trollface:

File: easy_profiler_core/CMakeLists.txt [hunk 2]
```diff
@@ -202,8 +206,11 @@ easy_define_target_option(easy_profiler EASY_OPTION_PROFILE_SELF EASY_OPTION_MEA
 easy_define_target_option(easy_profiler EASY_OPTION_PROFILE_SELF_BLOCKS_ON EASY_OPTION_STORAGE_EXPAND_BLOCKS_ON)
 easy_define_target_option(easy_profiler EASY_OPTION_IMPLICIT_THREAD_REGISTRATION EASY_OPTION_IMPLICIT_THREAD_REGISTRATION)
 if (WIN32)
-    easy_define_target_option(easy_profiler EASY_OPTION_EVENT_TRACING EASY_OPTION_EVENT_TRACING_ENABLED)
-    easy_define_target_option(easy_profiler EASY_OPTION_LOW_PRIORITY_EVENT_TRACING EASY_OPTION_LOW_PRIORITY_EVENT_TRACING)
+    if (${CMAKE_SYSTEM_NAME} STREQUAL "WindowsStore")
```
Comment (mathieumallet) [inline easy_profiler_core/CMakeLists.txt]: instead of having an empty block maybe it should be something like `if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "WindowsStore")` and no `else()`

File: easy_profiler_core/CMakeLists.txt [hunk 3]
```diff
@@ -202,8 +206,11 @@ easy_define_target_option(easy_profiler EASY_OPTION_PROFILE_SELF EASY_OPTION_MEA
...
+    else()
+        easy_define_target_option(easy_profiler EASY_OPTION_EVENT_TRACING EASY_OPTION_EVENT_TRACING_ENABLED)
+        easy_define_target_option(easy_profiler EASY_OPTION_LOW_PRIORITY_EVENT_TRACING EASY_OPTION_LOW_PRIORITY_EVENT_TRACING)
```
Comment (mathieumallet) [inline easy_profiler_core/CMakeLists.txt:213]: this didn't work on UWP? I haven't actually tried using it so I don't know how useful it is in the first place

File: easy_profiler_core/CMakeLists.txt [hunk 4]
```diff
@@ -219,7 +226,10 @@ easy_define_target_option(easy_profiler EASY_OPTION_PREDEFINED_COLORS EASY_OPTIO
 # Add platform specific compile options:
 if (UNIX)
     target_compile_options(easy_profiler PRIVATE -Wall -Wno-long-long -Wno-reorder -Wno-braced-scalar-init -pedantic)
-    target_link_libraries(easy_profiler pthread)
+    if (NOT ANDROID)
+    	#Android uses bionic which has its own implementation of pthread
```
Comment (mathieumallet) [inline easy_profiler_core/CMakeLists.txt]: tab instead of spaces

File: easy_profiler_core/include/easy/easy_socket.h [hunk 5]
```diff
@@ -94,6 +96,9 @@ class PROFILER_API EasySocket EASY_FINAL
     struct hostent* m_server = nullptr;
     struct sockaddr_in m_serverAddress;
 
+#if defined(__ORBIS__)
```
Comment (inantop) [inline easy_profiler_core/include/easy/easy_socket.h]: hmm looks like this is unused in the final version will delete

File: easy_profiler_core/CMakeLists.txt [hunk 6]
```diff
@@ -316,6 +312,7 @@ install(
 
 install(
     EXPORT "${targets_export_name}"
+    NAMESPACE "easy_profiler::"
```
Comment (mathieumallet) [inline easy_profiler_core/CMakeLists.txt:316]: what does this do?

File: easy_profiler_core/CMakeLists.txt (hunk 2 above)
Comment (inantop) [reply easy_profiler_core/CMakeLists.txt]: I actually couldn't get the `NOT` to work but I'll give it another go, maybe I was doing `NOT STREQUAL`

File: easy_profiler_core/CMakeLists.txt (hunk 3 above)
Comment (inantop) [reply easy_profiler_core/CMakeLists.txt:213]: iirc the APIs didn't exist

File: easy_profiler_core/include/easy/easy_socket.h [hunk 7]
```diff
@@ -94,6 +96,9 @@ class PROFILER_API EasySocket EASY_FINAL
...
+    int32_t youi_receiveFlag;
```
Comment (patrickyoui) [inline easy_profiler_core/include/easy/easy_socket.h]: What was this added for?

File: easy_profiler_core/current_thread.h [hunk 8]
```diff
@@ -47,20 +47,24 @@ The Apache License, Version 2.0 (the "License");
 
 #ifdef _WIN32
 # include <Windows.h>
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) or defined(__ORBIS__)
```
Comment (mathieumallet) [inline easy_profiler_core/current_thread.h]: would be clearer to `#elif defined(__ORBIS__)` and duplicate the `#include <pthread.h>`

File: easy_profiler_core/CMakeLists.txt (hunk 6 above)
Comment (inantop) [reply easy_profiler_core/CMakeLists.txt:316]: I don't remember the explanation given to me but we refer to these libraries with their namespaced names (maybe to avoid conflict with system libraries if I had to hazard a guess), so I had to add this to make the YiAddThirdpartyLibraries.cmake look consistent at least

File: easy_profiler_core/current_thread.h [hunk 9]
```diff
@@ -47,20 +47,24 @@ The Apache License, Version 2.0 (the "License");
...
 # include <pthread.h>
+#if defined (__APPLE__)
```
Comment (mathieumallet) [inline easy_profiler_core/current_thread.h]: technically this and the next two lines should be indented since it's within the `#elif`

File: easy_profiler_core/current_thread.h [hunk 10]
```diff
@@ -47,20 +47,24 @@ The Apache License, Version 2.0 (the "License");
...
+#  include <sys/syscall.h>
+# endif
 #endif
 
 inline profiler::thread_id_t getCurrentThreadId()
 {
 #ifdef _WIN32
     return (profiler::thread_id_t)::GetCurrentThreadId();
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) or defined(__ORBIS__) or defined(__pnacl__) || defined(__native_client__)
```
Comment (mathieumallet) [inline easy_profiler_core/current_thread.h]: mixed `or` and `||` -- should stick with one of them

File: easy_profiler_core/easy_socket.cpp [hunk 11]
```diff
@@ -53,7 +57,19 @@ limitations under the License.
...
+# endif 
+# if defined(__pnacl__) || defined(__native_client__)
+#  include <sys/time.h>
+#  include <sys/fcntl.h>
+#  include <sys/select.h>
+# elif defined(__ORBIS__)
+#  include <sys/ioccom.h>
+# else
+#  include <sys/ioctl.h>
+#endif
```
Comment (mathieumallet) [inline easy_profiler_core/easy_socket.cpp]: missing indent

File: easy_profiler_core/easy_socket.cpp [hunk 12]
```diff
@@ -53,7 +57,19 @@ limitations under the License.
 # endif
 #else
 # include <errno.h>
-# include <sys/ioctl.h>
+# ifndef FIONBIO
+//According to https://stackoverflow.com/questions/1150635/unix-nonblocking-i-o-o-nonblock-vs-fionbio the O_NONBLOCK and FIONBIO flags are treated equivalent under POSIX, but FIONBIO is not defined on PS4 or on tizen
+#  define FIONBIO O_NONBLOCK
+# endif 
+# if defined(__pnacl__) || defined(__native_client__)
+#  include <sys/time.h>
```
Comment (mathieumallet) [inline easy_profiler_core/easy_socket.cpp]: unsorted includes 🚲

File: easy_profiler_core/include/easy/easy_socket.h [hunk 13]
```diff
@@ -50,7 +50,9 @@ The Apache License, Version 2.0 (the "License");
 // Unix
 # include <sys/types.h>
 # include <sys/socket.h>
+#if !defined(__ORBIS__)
```
Comment (mathieumallet) [inline easy_profiler_core/include/easy/easy_socket.h]: indent (and next 2 lines) 🚲

File: easy_profiler_core/include/easy/easy_socket.h (hunk 7 above)
Comment (mathieumallet) [inline easy_profiler_core/include/easy/easy_socket.h]: `m_youi_receiveFlag`?

File: easy_profiler_core/profile_manager.cpp [hunk 14]
```diff
@@ -1444,10 +1448,14 @@ void ProfileManager::listen(uint16_t _port)
                     socket.setReceiveTimeout(0);
                     clear_sstream(os);
                 }
-                else if (dumpingResult.wait_for(std::chrono::milliseconds(0)) == std::future_status::ready)
+#if defined(__pnacl__) || defined(__native_client__)
+                else //will force a blocking call at std::future::get below -- the wait_for was not working on tizen.
+#else
+		else if (dumpingResult.wait_for(std::chrono::milliseconds(0)) == std::future_status::ready)
```
Comment (mathieumallet) [inline easy_profiler_core/profile_manager.cpp]: tabs instead of spaces

File: easy_profiler_core/profiler.cpp [hunk 15]
```diff
@@ -62,6 +62,10 @@
 # endif
 #endif
 
+#if defined(_WIN32) && !defined(_UWP)
+#define USE_TRACING
```
Comment (mathieumallet) [inline easy_profiler_core/profiler.cpp]: indent

File: profiler_gui/blocks_graphics_view.cpp [hunk 16]
```diff
@@ -2179,6 +2180,10 @@ void BlocksGraphicsView::onIdleTimeout()
     decltype(scenePos) pos(m_offset + scenePos.x() / m_scale, scenePos.y());
 
     // Try to select one of context switches or items
```
Comment (mathieumallet) [inline profiler_gui/blocks_graphics_view.cpp]: comment should be after the `isActiveWindow` check

**Review by mathieumallet — APPROVED** [review]
+1 w/ comments

File: easy_profiler_core/include/easy/easy_socket.h (hunk 7 above)
Comment (mathieumallet) [reply easy_profiler_core/include/easy/easy_socket.h]: doesn't look like it's used anywhere
